cabal-install/src/Distribution/Client/ProjectPlanning.hs

```diff
@@ -1218,6 +1218,7 @@ planPackages verbosity comp platform solver SolverSettings{..}
     -- TODO: long-term, this compatibility matrix should be
     --       stored as a field inside 'Distribution.Compiler.Compiler'
     setupMinCabalVersionConstraint
+      | isGHC, compVer >= mkVersion [9,8]  = mkVersion [3,10]
```

Comment:
Notice that in the list below there's no precedent for Cabal's version to stagger. So, your change needs some reflection, perhaps. Notice the comment above:

```
-- Moreover, lib:Cabal generally only supports the interface of
    -- current and past compilers; 
```
do we make Cabal 3.10.1 support 9.8 when running with cabal-install 3.10.2 here? Will that work?

```
in fact recent lib:Cabal versions
    -- will warn when they encounter a too new or unknown GHC compiler
    -- version (c.f. https://github.com/haskell/cabal/issues/415). To avoid running into unsupported
    -- configurations we encode the compatibility matrix as lower
    -- bounds on lib:Cabal here (effectively corresponding to the
    -- respective major Cabal version bundled with the respective GHC
    -- release).
```
So, this `effectively` part will not be true anymore if GHC 9.8 bundles Cabal 3.12? (Will it? I'm a bit out of loop here.)

Reply:
> do we make Cabal 3.10.1 support 9.8 when running with cabal-install 3.10.2 here? Will that work?

I wasn't really thinking about doing this. For me that would be 3.10.2 that would support 9.8

Reply:
If specifying a minor version here works (again, no precedent), then indeed we don't have to think whether 3.10.1 misses anything that is required for GHC 9.8 support (I think not, it's a bugfix release only, but I may be wrong). And it's generally more accurate regarding our intent and regarding which Cabal version is bundled where, which future tools may for whatever reason like to know.